resolvers/index.js

```diff
@@ -42,28 +38,34 @@ module.exports = {
         if (customerPaymentIds) {
           paymentIds.push(...customerPaymentIds);
         }
-
-        if (customerTariffPlansIds) {
-          tariffPlanIds.push(customerTariffPlansIds);
-        }
       });
 
-      let [meterReadings, payments, tariffPlans] = await Promise.all([
+      let [meterReadings, payments] = await Promise.all([
         getMeterReadingsandInvoicesByIds(meterReadingIds),
         getPaymentsByIds(paymentIds),
-        getTariffPlansByIds(tariffPlanIds),
       ]);
 
-      matchCustomersWithReadingsPaymentsAndTariff(
-        customers,
-        meterReadings,
-        payments,
-        tariffPlans
-      );
+      matchCustomers(customers, meterReadings, payments);
 
       siteRecord.fields.Customers = customers;
     }
 
+    const financialSummaryIds = siteRecord.fields["Fainancial Summaries"];
+    let financialSummaries = [];
+    if (financialSummaryIds) {
+      financialSummaries = await getFinancialSummariesByIds(
+        financialSummaryIds
+      );
+    }
+    siteRecord.fields.FinancialSummaries = financialSummaries;
+
+    const tariffPlanIds = siteRecord.fields["Tariff Plans"];
+    let tariffPlans = [];
+    if (tariffPlanIds) {
+      tariffPlans = await getTariffPlansByIds(tariffPlanIds);
+    }
+    siteRecord.fields.TariffPlans = tariffPlans;
+
```

Comment:
We should probably structure this code so that all API calls are awaited on together (financialSummaries and Tariff Plans should be included in the Promise.all await) in order to speed things up. This way we only have to await 1 round of API calls

Reply:
Should I combine it with line 43 then?

Reply:
Yeah that works. Logic looks good i'd rather just not wait the extra API call since it's unnecessary. Combining with line 43 works as a solution.